File: BinarySearch/FindMinRotatedArray.py
class Solution(object):
    def findMin(self, nums):
        i, j = 0 , len(nums)-1
        mini = nums[0]
        while i <= j:
            mid = int((i+j+1)/2)
            if nums[0] < nums[mid]:
                i = mid+1
            else:
                j = mid-1
                mini = min(nums[mid],mini)
        return mini

File: BinarySearch/FindMinRotatedArray.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int findMin(vector<int>& nums) {
        int i = 0;
        int j = nums.size() - 1;
        int mid;
        int mini = nums[j];
        while(i <= j)
        {
            mid = (j+i+1) / 2;
            if(nums[0]<nums[mid])
                i = mid+1;
            else
            {
                mini = min(nums[mid],mini);
                j = mid-1;
            }
        }
        return min(nums[0],mini);
    }
};